From 88f4404bfea0f0a086e1ae6cf939132160c9924f Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Thu, 28 Mar 2013 21:44:43 -0400
Subject: [PATCH] hrtimer: fix compile failure for when HRT are disabled
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With "# CONFIG_HIGH_RES_TIMERS is not set" in .config, one will see:

kernel/hrtimer.c: In function ‘__hrtimer_start_range_ns’:
kernel/hrtimer.c:1045:7: error: implicit declaration of function ‘hrtimer_rt_defer’ [-Werror=implicit-function-declaration]
kernel/hrtimer.c: At top level:
kernel/hrtimer.c:1416:12: error: static declaration of ‘hrtimer_rt_defer’ follows non-static declaration
kernel/hrtimer.c:1045:10: note: previous implicit declaration of ‘hrtimer_rt_defer’ was here
cc1: some warnings being treated as errors
make[2]: *** [kernel/hrtimer.o] Error 1

Move the prototype outside of the block of code which is conditional
on high res timers.

Signed-off-by: Paul Gortmaker <paul.gortmaker@xxxxxxxxxxxxx>

diff --git a/kernel/hrtimer.c b/kernel/hrtimer.c
index 13127f0..c534153 100644
--- a/kernel/hrtimer.c
+++ b/kernel/hrtimer.c
@@ -630,7 +630,6 @@ static int hrtimer_reprogram(struct hrtimer *timer,
 }
 
 static void __run_hrtimer(struct hrtimer *timer, ktime_t *now);
-static int hrtimer_rt_defer(struct hrtimer *timer);
 
 /*
  * Initialize the high resolution related parts of cpu_base
@@ -748,6 +747,8 @@ static inline int hrtimer_reprogram(struct hrtimer *timer,
 
 #endif /* CONFIG_HIGH_RES_TIMERS */
 
+static int hrtimer_rt_defer(struct hrtimer *timer);
+
 /*
  * Clock realtime was set
  *
-- 
1.8.3.1

